linux_os/guide/system/network/network-nftables/var_nftables_base_chain_hooks.var

```diff
@@ -0,0 +1,32 @@
+documentation_complete: true
+
+title: 'Nftables Base Chain Hooks'
+
+description: |-
+   The possible hooks which can be used to configure the base chain are:
+   <tt>ingress</tt> (only in netdev family since Linux kernel 4.2, and inet family since Linux kernel 5.10): 
+   sees packets immediately after they are passed up from the NIC driver, before even prerouting. 
+   <tt>prerouting</tt> sees all incoming packets, before any routing decision has been made. 
+   Packets may be addressed to the local or remote systems.
+   <tt>input</tt> sees incoming packets that are addressed to and have now been routed 
+   to the local system and processes running there.
+   <tt>forward</tt> sees incoming packets that are not addressed to the local system.
+   <tt>output</tt> sees packets that originated from processes in the local machine.
+   <tt>postrouting</tt> sees all packets after routing, just before they leave the 
+   local system.
+
+type: string
+
+operator: equals
+
+interactive: true
+
+options:
+    default: arr_chain_hooks
+    ingress: ingress 
+    prerouting: prerouting
+    input: input
+    forward: forward
+    output: output
+    postrouting: postrouting
+    arr_chain_hooks: "input,forward,output"
```

Comment:
Hello @marcusburghardt,  
When I read documentation about nftables (see this link for example https://wiki.nftables.org/wiki-nftables/index.php/Configuring_tables) - I did not find filter table. I think that we can specify any name as a table name, but important parts are families of the tables, and also we can specify any name for the chain name, but important parts are  
chain types - filter, route, nat
chain hooks - ingress, prerouting, input, forward, output, postrouting 
(https://wiki.nftables.org/wiki-nftables/index.php/Configuring_chains#Base_chain_hooks) 
And because of that when I created variables I used this concept.
If I use filter_chain_hooks, probably we will need to define, route_chain_hooks, and nat_chain_hooks
Have a nice day
Rumen

Reply:
I see @rumch-se . My point is about the scalability of options names in this variable. Supposing we need an additional option in the future to cover specifically `prerouting, input, forward, output, postrouting`. How would we name this new option? That is the idea behind my suggestion, to have a generic enough but also meaningful standard for options names so the addition of future options are better aligned.

The "filter" was just a suggestion probably influenced by my previous experience in `iptables`. But was just a suggestion. Since the options names follow a scable standard, I am fine.

Reply:
Hello @marcusburghardt 

Here an example, which shows a case with 6 chains in table my_table which belongs to family inet. Probably some of chains are not correct, but I wanted to show how we will work if I need more than 3 chains. Because of that I need arrays and and used arr_ as a prefix to show that.
I used variable var_nftables_base_chain_number - because you can use it to control number of chains you want. For example you can have 6, or 10 elements into var_nftables_base_chain_names, var_nftables_base_chain_types, etc., but you can want to create only 2 chains. 

var_nftables_family: inet
var_nftables_table:  my_table

var_nftables_base_chain_number.var:6

var_nftables_base_chain_names.var:
arr_chain_names: "input,input,forward,forward,output,output"

var_nftables_base_chain_types.var
arr_chain_types: "filter,route,nat,filter,route,nat"

var_nftables_base_chain_hooks.var:
arr_chain_hooks: "ingress,prerouting,input,forward,output,postrouting"

var_nftables_base_chain_priorities.var:
arr_chain_priorities: "0,50,100,225,-200,-300"

var_nftables_base_chain_policies.var:
arr_chain_policies: "accept,drop,accept,accept,drop,accept"


add chain $var_nftables_family $var_nftables_table ${arr_chnames[$i]} { type ${arr_chtypes[$i]} hook ${arr_chhooks[$i]} priority ${arr_chpriorities[$i]} ; policy ${arr_chpolicies[$i]} ; }"
 
The result will be: 
1)add chain my_table inet input { type filter hook ingress priority 0 policy accept }
2)add chain my_table inet input { type route hook prerouting priority 50 policy drop } 
3)add chain my_table inet forward { type nat hook input priority 100 policy accept }
4)add chain my_table inet forward { type filter hook forward priority 225 policy accept }
5)add chain my_table inet output { type route hook output priority -200 policy drop }
6)add chain my_table inet output { type nat hook postrouting priority -300 policy accept }

Have a nice day
Rumen

Reply:
Ok. I personally don't think `arr` is a good prefix but I have no strong opinion on that or a better generic suggestion which are not opinionated. I can be fine with the prefix you choose. Please, take a look in my other comments too so we can move forward with this PR. Thanks.

Reply:
Hello @marcusburghardt
The prefix arr_ was removed
Have a nice day 
Rumen